osgar/drivers/cortexpilot.py

```diff
@@ -0,0 +1,125 @@
+"""
+  Driver for robot Robik from cortexpilot.com
+"""
+
+import struct
+import math
+
+from osgar.node import Node
+from osgar.bus import BusShutdownException
+
+
+# CPR = 9958 (ticks per revolution)
+# wheel diameter D = 395 mm
+# 1 Rev = 1241 mm
+ENC_SCALE = 1.241/9958
+
+
+class Cortexpilot(Node):
+    def __init__(self, config, bus):
+        super().__init__(config, bus)
+
+        self._buf = b''
+
+        # commands
+        self.desired_speed = 0.0  # m/s
+        self.desired_angular_speed = 0.0
+        self.cmd_flags = 0x41 #0x40  # 0 = remote steering, PWM OFF, laser ON, TODO
+
+        # status
+        self.emergency_stop = None  # uknown state
+        self.pose = (0.0, 0.0, 0.0)  # x, y in meters, heading in radians (not corrected to 2PI)
+        self.flags = None
+        self.voltage = None
+        self.last_encoders = None
+        self.yaw = None
+
+    def send_pose(self):
+        x, y, heading = self.pose
+        self.publish('pose2d', [round(x*1000), round(y*1000),
+                                round(math.degrees(heading)*100)])
+
+    def query_version(self):
+        ret = bytes([0, 0, 3, 0x1, 0x01])
+        checksum = sum(ret) & 0xFF
+        return ret + bytes([256-checksum])
+
+    def create_packet(self):
+        if self.yaw is None:
+            self.yaw = 0.0  # hack!
+        packet = struct.pack('<ffI', self.desired_speed,
+#                             self.desired_angular_speed, self.cmd_flags)
+                             self.yaw, self.cmd_flags)
+        assert len(packet) < 256, len(packet)  # just to use LSB only
+        ret = bytes([0, 0, len(packet) + 2 + 1, 0x1, 0x0C]) + packet
+        checksum = sum(ret) & 0xFF
+        return ret + bytes([256-checksum])
+
+    def get_packet(self):
+        """extract packet from internal buffer (if available otherwise return None"""
+        data = self._buf
+        if len(data) < 5:
+            return None
+        high, mid, low = data[:3]  # packet length
+        assert high == 0, high  # all messages < 65535 bytes
+        size = 256 * mid + low + 3  # counting also 3 bytes of len header
```

Comment:
I don't see where this `+3` comes from. In `create_packet`
```python
ret = bytes([0, 0, len(packet) + 2 + 1, 0x1, 0x0C]) + packet
```
there is also `+3` but I see only 2 additional bytes being put there.  Also
```python
if len(data) < 5:
```
suggest the header is 3 bytes size + 2 bytes of something. What is the reasoning here?